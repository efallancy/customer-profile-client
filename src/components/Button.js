import styled from 'styled-components';

export const PrimaryButton = styled.button`
  font-size: 20px;
  background-color: #13D0A7;
  border: none;
  border-radius: 20px;
  box-sizing: border-box;
  color: #FFFFFF;
  padding: 0.5rem;
  min-width: 100px;
  margin: 0.5rem 0;
  outline: none;
  &:hover {
    cursor: ${props => props.disabled ? 'not-allowed' : 'pointer'};
  }
`;

export const SecondaryButton = PrimaryButton.extend`
  background-color: #FFFFFF;
  border: 2px solid #13D0A7;
  color: #13D0A7;
`;

export default PrimaryButton;