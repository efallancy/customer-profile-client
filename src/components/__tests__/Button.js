import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { PrimaryButton, SecondaryButton } from '../Button';

describe('Button', () => {
  describe('Primary', () => {
    it('renders correctly', () => {
      const button = renderer.create(<PrimaryButton />).toJSON();
      expect(button).toMatchSnapshot();
    });

    it('renders correctly when disabled', () => {
      const button = renderer.create(<PrimaryButton disabled />).toJSON();
      expect(button).toMatchSnapshot();
    });
  });

  describe('Secondary', () => {
    it('renders correctly', () => {
      const button = renderer.create(<SecondaryButton />).toJSON();
      expect(button).toMatchSnapshot();
    });

    it('renders correctly when disabled', () => {
      const button = renderer.create(<SecondaryButton disabled />).toJSON();
      expect(button).toMatchSnapshot();
    });
  });
});
