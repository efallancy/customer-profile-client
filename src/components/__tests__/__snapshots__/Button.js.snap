// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Primary renders correctly 1`] = `
.c0 {
  font-size: 20px;
  background-color: #13D0A7;
  border: none;
  border-radius: 20px;
  box-sizing: border-box;
  color: #FFFFFF;
  padding: 0.5rem;
  min-width: 100px;
  margin: 0.5rem 0;
  outline: none;
}

.c0:hover {
  cursor: pointer;
}

<button
  className="c0"
/>
`;

exports[`Button Primary renders correctly when disabled 1`] = `
.c0 {
  font-size: 20px;
  background-color: #13D0A7;
  border: none;
  border-radius: 20px;
  box-sizing: border-box;
  color: #FFFFFF;
  padding: 0.5rem;
  min-width: 100px;
  margin: 0.5rem 0;
  outline: none;
}

.c0:hover {
  cursor: not-allowed;
}

<button
  className="c0"
  disabled={true}
/>
`;

exports[`Button Secondary renders correctly 1`] = `
.c0 {
  font-size: 20px;
  background-color: #13D0A7;
  border: none;
  border-radius: 20px;
  box-sizing: border-box;
  color: #FFFFFF;
  padding: 0.5rem;
  min-width: 100px;
  margin: 0.5rem 0;
  outline: none;
  background-color: #FFFFFF;
  border: 2px solid #13D0A7;
  color: #13D0A7;
}

.c0:hover {
  cursor: pointer;
}

<button
  className="c0"
/>
`;

exports[`Button Secondary renders correctly when disabled 1`] = `
.c0 {
  font-size: 20px;
  background-color: #13D0A7;
  border: none;
  border-radius: 20px;
  box-sizing: border-box;
  color: #FFFFFF;
  padding: 0.5rem;
  min-width: 100px;
  margin: 0.5rem 0;
  outline: none;
  background-color: #FFFFFF;
  border: 2px solid #13D0A7;
  color: #13D0A7;
}

.c0:hover {
  cursor: not-allowed;
}

<button
  className="c0"
  disabled={true}
/>
`;
